### ISSUE TITLE
使用imageListToImageBatch节点图片多的时候效率较低

### ISSUE THREAD

**Issue by civen-cn** [yolain/ComfyUI-Easy-Use#421]
image1 = images[0]
      for image2 in images[1:]:
        if image1.shape[1:] != image2.shape[1:]:
          image2 = comfy.utils.common_upscale(image2.movedim(-1, 1), image1.shape[2], image1.shape[1], "lanczos",
                                              "center").movedim(1, -1)
        image1 = torch.cat((image1, image2), dim=0)

发现应该是反复torch.cat爆内存的情况，通常情况images输入格式都一致，没必要做common_upscale这些操作。把上面代码直接改成

**Comment by civen-cn** [comment]
image1 = torch.cat(images, dim=0)
即可。
或者另做一个简单操作的imageListToImageBatch

**Comment by yolain** [comment]
应该是缩进问题

**Comment by AndrewZhuCC** [comment]
80ede24 这里改掉之后变成了只会合并第一张和最后一张图片，如果有两张以上的话，那么会导致中间的图片都被忽略 @yolain

**Comment by yolain** [comment]
> 80ede24 这里改掉之后变成了只会合并第一张和最后一张图片，如果有两张以上的话，那么会导致中间的图片都被忽略 @yolain 

我重新修复一下

**Comment by yolain** [comment]
原来写的是对的。那天看错了 我以为是因为缩进导致节点报错了。节点太多看乱了🥲impact pack那边已经有一个了 当时在easyuse加这个是为了解决当尺寸不一致时会报错

**Comment by yolain** [comment]
原来写的是对的。那天看错了 我以为是因为缩进导致节点报错了。节点太多看乱了🥲impact pack那边已经有一个了 当时在easyuse加这个是为了解决当尺寸不一致时会报错。缩放也只会在尺寸不同的时候才会进行，内存异常可能是图太大或者数量太多导致的

**Comment by civen-cn** [comment]
> 原来写的是对的。那天看错了 我以为是因为缩进导致节点报错了。节点太多看乱了, 那天又刚好在准备更新新的内容，没细测 抱歉🥲impact pack那边已经有一个了 当时在easyuse加这个是为了解决当尺寸不一致时会报错。缩放也只会在尺寸不同的时候才会进行，内存异常可能是图太大或者数量太多导致的

list每个都做 image1 = torch.cat((image1, image2), dim=0) 更容易爆内存，这边试出来就是这样的。直接

**Comment by civen-cn** [comment]
4090跑的，
直接torch.cat(images, dim=0)的用时仅3s
![image](https://github.com/user-attachments/assets/49c398b7-015e-41f4-a02d-eb98f05e3e54)

逐个cat用了好久都出不来。貌似内存没有增（上次我出的图可能还真存在大小不一的情况），就是慢
增加计数打印情况：
[2024-10-25 16:24:19.736] 0
9s
[2024-10-25 16:24:28.295] 100
25s
[2024-10-25 16:24:53.187] 200
41s
[2024-10-25 16:25:34.383] 300
57s
[2024-10-25 16:26:31.878] 400
73s
[2024-10-25 16:27:45.658] 500
90s
[2024-10-25 16:29:15.822] 600
107s
[2024-10-25 16:31:02.193] 700
122s
[2024-10-25 16:33:04.802] 800
139s
[2024-10-25 16:35:23.646] 900
155s
[2024-10-25 16:37:58.762] 1000
172s
[2024-10-25 16:40:50.775] 1100
189s
[2024-10-25 16:43:59.917] 1200
